exporters/otlp/otlpmetric/internal/transform/metricdata.go

```diff
@@ -198,6 +202,62 @@ func HistogramDataPoints[N int64 | float64](dPts []metricdata.HistogramDataPoint
 	return out
 }
 
+// ExponentialHistogram returns an OTLP Metric_ExponentialHistogram generated from h. An error is
+// returned with a partial Metric_Histogram if the temporality of h is
+// unknown.
```

Comment:
I do not see any "partial Metric_Histogram"

```suggestion
// returned if the temporality of h is unknown.
```

If I am not mistaken then same issue exists in other places in this file. I created https://github.com/open-telemetry/opentelemetry-go/pull/4242